cdap-app-fabric/src/main/java/io/cdap/cdap/internal/app/preview/RemotePreviewRequestFetcher.java

```diff
@@ -62,6 +62,10 @@
     HttpResponse httpResponse = remoteClientInternal.execute(request);
     if (httpResponse.getResponseCode() == 200) {
       PreviewRequest previewRequest = GSON.fromJson(httpResponse.getResponseBodyAsString(), PreviewRequest.class);
+      if (previewRequest != null && previewRequest.getPrincipal() != null) {
```

Comment:
Is this confirmed to be the same thread in which the preview run is executed?

Reply:
Yes. Preview request pull and execution is done in the same thread.